### DISCUSSION TITLE
FM's clear book cache reset appearance

### DISCUSSION THREAD

**Discussion by mergen3107** [koreader/koreader#9971] [Q&A]
So in the FM under the mosaic settings there is an option to wipe the cache for books.

Is it intended that this action also reset the appearance from detailed with cover (or else) to simple list without covers?
It also unchecks the series appearance in books.

**Comment by poire-z** [comment]
> Is it intended that this action also reset the appearance 

Not super intended, but just a side effect of deleting the database file, where the cover display mode settings are also stored (at the time, I wanted this plugin to be self contained and to not pollute the main settings - and to have an option to delete the file and start from scratch :)

`text = _("Are you sure that you want to delete cover and metadata cache?\n(This will also reset your display mode settings.)")`

**Reply by mergen3107** [reply]
Ok, thank you :) I just keep forgetting to turn on Series appearance after cache overhaul xD

and I do cache overhaul from time to time because when I do “delete cache for deleted books” the file size stays the same. So it keeps going up, and when it is around 50 MB I delete it all and start from scratch.

**Reply by poire-z** [reply]
> the file size stays the same. So it keeps going up

There's that other menu item for that: `Compact cache database` (which creates a new db and copy all the not-deleted content to that new db, which should end up smaller).

**Reply by mergen3107** [reply]
Oh, ok! I thought “compact” meant “compressed”. One time a long time ago it didn’t work well for me - it crashed after compacting. Now it is fine, let me test for a while. 

How is “Compact” is different from “Prune cache of removed database”?

This submenu overall has a few UX questions so far:
1) Neither of menus (Cache size, Prune, Compact, Delete) have explanation on long press. 
2) Once Compact is done, it reports well on the new size. However, the menu item isn’t yet refreshed. It’ll refresh if submenu is closed and opened again. 
3) And the above question about how Compact is different from Prune.

**Reply by poire-z** [reply]
> One time a long time ago it didn’t work well for me - it crashed after compacting.

(I think I remember it - or it was somebody else.)
When it is compacting, it may needs both the space, for the old db, and the new one being created and fed data - only once this is done the old db is deleted. So, if not enough space, and not enough hole in the old db so the new one get small enough: failure/crash.

> 1 Neither of menus (Cache size, Prune, Compact, Delete) have explanation on long press.

Yep, didn't think about adding some for these technical menu items :/

>2 Once Compact is done, it reports well on the new size. However, the menu item isn’t yet refreshed. It’ll refresh if submenu is closed and opened again.

Yep, didn't think/bother about making all that smooth for these technical menu items not often used :)

> 3 And the above question about how Compact is different from Prune.

Prune deletes the data in the DB - just making the space they took available for other future data: so it doesn't reduce the db size (it keeps the hole), but will keep it from growing with the future insertions (which will be inserted in the hole) for some time.
Compact get rid of the holes while making a new file, so you end up seeing the file smaller. But then future insertions will make the new file immediately grow (as they are no available hole to fit new data in).

**Reply by NiLuJe** [reply]
> So, if not enough space, and not enough hole in the old db so the new one get small enough: failure/crash.

There *may* also be a RAM requirement at play, IIRC (I may be wrong there), which could be problematic on memory-constrained devices ;).

**Reply by mergen3107** [reply]
Thank you @poire-z ! Now I understand it much better. 

I’ll try to see if I can add translatable long-press explanations for these items using your messages above. 

@offset-torque
Do you think this information is worth adding to the manual? Or just the long-press messages would suffice?

**Comment by offset-torque** [comment]
>  **mergen3107** - Do you think this information is worth adding to the manual? Or just the long-press messages would suffice?

This looks like a (well-) hidden troubleshooting option. I think it  deserves a mention.
Can I get a short description for: 
**"In which cases a user might need to use this option ?"**

**Reply by poire-z** [reply]
> "In which cases a user might need to use this option ?"

"If you are anal retentive, and remove a lot of books or rename directories, you may want to clean your book cover cache database and reclaim some space" :)

**Reply by offset-torque** [reply]
I see. So it is just about space. I thought when database reaches a size/complexity it might get corrupted/confused and users get wrong/missing covers or some problem like that.
Adding this info to the **TIPS** section of the guide then (instead of **TROUBLESHOOTING**).
